test/strategy/rysk/RyskStrategy.spec.ts

```diff
@@ -0,0 +1,639 @@
+
+    RyskStrategyFactory = await ethers.getContractFactory('RyskStrategy');
+
+    strategy = await RyskStrategyFactory.deploy(
+      vault.address,
+      admin.address,
+      keeper.address,
+      ryskLqPool.address,
+      underlying.address,
+    );
+
+    await strategy.connect(admin).grantRole(MANAGER_ROLE, manager.address);
+
+    await vault.setStrategy(strategy.address);
+
+    await underlying
+      .connect(admin)
+      .approve(vault.address, constants.MaxUint256);
+  });
+
+  describe('#constructor', () => {
+    it('reverts if the admin is address(0)', async () => {
+      await expect(
+        RyskStrategyFactory.deploy(
+          vault.address,
+          constants.AddressZero,
+          keeper.address,
+          ryskLqPool.address,
+          underlying.address,
+        ),
+      ).to.be.revertedWith('StrategyAdminCannotBe0Address');
+    });
+
+    it('reverts if the keeper is address(0)', async () => {
+      await expect(
+        RyskStrategyFactory.deploy(
+          vault.address,
+          admin.address,
+          constants.AddressZero,
+          ryskLqPool.address,
+          underlying.address,
+        ),
+      ).to.be.revertedWith('StrategyKeeperCannotBe0Address');
+    });
+
+    it('reverts if the Rysk liquidity pool is address(0)', async () => {
+      await expect(
+        RyskStrategyFactory.deploy(
+          vault.address,
+          admin.address,
+          keeper.address,
+          constants.AddressZero,
+          underlying.address,
+        ),
+      ).to.be.revertedWith('RyskLiquidityPoolCannotBe0Address');
+    });
+
+    it('reverts if underlying is address(0)', async () => {
+      await expect(
+        RyskStrategyFactory.deploy(
+          vault.address,
+          admin.address,
+          keeper.address,
+          ryskLqPool.address,
+          constants.AddressZero,
+        ),
+      ).to.be.revertedWith('StrategyUnderlyingCannotBe0Address');
+    });
+
+    it('reverts if vault does not have IVault interface', async () => {
+      await expect(
+        RyskStrategyFactory.deploy(
+          manager.address, // vault param
+          admin.address,
+          keeper.address,
+          ryskLqPool.address,
+          underlying.address,
+        ),
+      ).to.be.revertedWith('StrategyNotIVault');
+    });
+
+    it('sets initial values as expected', async () => {
+      expect(await strategy.vault()).to.eq(vault.address);
+      expect(await strategy.ryskLqPool()).to.eq(ryskLqPool.address);
+      expect(await strategy.underlying()).to.eq(underlying.address);
+
+      expect(await strategy.isSync()).to.be.false;
+
+      expect(await strategy.hasRole(DEFAULT_ADMIN_ROLE, admin.address)).to.be
+        .true;
+      expect(await strategy.hasRole(MANAGER_ROLE, vault.address)).to.be.true;
+
+      expect(
+        await underlying.allowance(strategy.address, ryskLqPool.address),
+      ).to.eq(constants.MaxUint256);
+    });
+  });
+
+  describe('#transferAdminRights', () => {
```

Comment:
don't we have these tests for the base strategy?

Reply:
yes, we do. This is a leftover since these tests were written before we introduced the base strategy contract. I'll remove them

Reply:
[removed](https://github.com/lindy-labs/sc_solidity-contracts/pull/152/commits/3d0782af21c18b94db8f2178ae5788d2012ca4cc)